main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/ONSdigital/dp-api-clients-go/renderer"
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+
+	"github.com/ONSdigital/dp-frontend-homepage-controller/config"
+	"github.com/ONSdigital/dp-frontend-homepage-controller/routes"
+	"github.com/ONSdigital/go-ns/server"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+)
+
+var (
+	// BuildTime represents the time in which the service was built
+	BuildTime string
+	// GitCommit represents the commit (SHA-1) hash of the service that is running
+	GitCommit string
+	// Version represents the version of the service that is running
+	Version string
+)
+
+func main() {
+	log.Namespace = "dp-frontend-homepage-controller"
+	cfg, err := config.Get()
+	ctx := context.Background()
+	if err != nil {
+		log.Event(ctx, "unable to retrieve service configuration", log.Error(err))
+		os.Exit(1)
+	}
+
+	log.Event(ctx, "got service configuration", log.Data{"config": cfg})
+
+	versionInfo, err := health.NewVersionInfo(
+		BuildTime,
+		GitCommit,
+		Version,
+	)
+
+	r := mux.NewRouter()
+
+	rend := renderer.New(cfg.RendererURL)
+
+	healthcheck := health.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+	if err = registerCheckers(ctx, &healthcheck); err != nil {
+		os.Exit(1)
+	}
+	routes.Init(ctx, r, healthcheck, rend)
+
+	healthcheck.Start(ctx)
+
+	s := server.New(cfg.BindAddr, r)
+	s.HandleOSSignals = false
+
+	log.Event(ctx, "Starting server", log.Data{"config": cfg})
+
+	go func() {
+		if err := s.ListenAndServe(); err != nil {
+			log.Event(ctx, "failed to start http listen and serve", log.Error(err))
+			return
+		}
+	}()
+
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, os.Kill)
+
+	<-stop
+
```

Comment:
There's an issue here I believe. AS I understand it, if there's an internal error with starting/running `s.ListenAndServe()` then that go routine will silently die but the outer go routine will still be running waiting for an os.Signal. Ie. the inner go routine hasn't got an error channel to return errors. I've done this a different way in my bulletin api that I think is more robust but I'd appreciate someone like @janderson2, @eldeal or some other more experienced go dev to clarify. Once my PR is approved I intend to get this into the repo generation templates.

NB. My bulletin API example…
main.go (service errors channel): https://github.com/ONSdigital/dp-bulletin-api/blob/feature/skeleton-api/main.go#L35
service.go (inner go routine passing back errors): https://github.com/ONSdigital/dp-bulletin-api/blob/feature/skeleton-api/service/service.go#L57

Reply:
You're absolutely right that the goroutine for ListenAndServe is failing silently and another channel is definitely a way of capturing these failures. In our case it might not be necessary though. If `ListenAndServe` fails, the `/health` endpoint will not be served, so the platform _will_ send an OS signal in short order. 

Given that in all apps the goroutine that is not being handled is only calling `ListenAndServe`, and we know the platform will kill things, I'd suggest right now it's not necessary to add the complexity of passing an additional channel down through function calls to pass this edge case back up. 

I'd have less concerns about the complexity in an app where the main package still contains all the logic, as I personally find the service abstraction complex so I'm hesitant to add more complexity of following channels through it. So I'd say there's a judgement call to be made as a team about where we want our complexity, and whether we think this is worth it for the edge case. I personally think not, but I also think if the team feels strongly about it then I'm happy to be persuaded.

Its a good spot in either case, as if we have other things in goroutines at this level we may need to revisit this pattern in future, where the platform isn't catching it.